docs/docs/refs-and-the-dom.md

```diff
@@ -50,9 +50,11 @@ class CustomTextInput extends React.Component {
 }
 ```
 
-React will call the `ref` callback with the DOM element when the component mounts, and call it with `null` when it unmounts.
+React will call the `ref` callback with the DOM element when the component mounts, and call it with `null` when it unmounts. Additionally, if the `ref` callback is defined as an inline function, it will get called twice during each render pass, first with `null` and then again with the DOM element. This is because internally, a new instance of the function is created with each render, so React needs to reset the old instance and set-up the new one. This can be avoided by defining the ref callback as a property on the class, but note that it shouldn't matter in the average `ref` use case.
```

Comment:
The important part is that while some readers will be new just to callback refs, for a growing number of readers over time this will be the first introduction to the concept of refs at all. And as such, throwing too many details on them will only confuse them.